politeiawww/userwww.go

```diff
@@ -47,32 +42,87 @@ func (p *politeiawww) isAdmin(w http.ResponseWriter, r *http.Request) (bool, err
 	return user.Admin, nil
 }
 
-// getSessionUUID returns the uuid address of the currently logged in user from
-// the session store.
-func (p *politeiawww) getSessionUUID(r *http.Request) (string, error) {
-	session, err := p.getSession(r)
+// getCookie returns the active cookie session.
+func (p *politeiawww) getCookie(r *http.Request) (*sessions.Session, error) {
+	return p.store.Get(r, www.CookieSession)
+}
+
+// getSession looks up the user session for the given request in the database.
+//
+// Please note that the session record in the database will be deleted and the
+// cookie invalidated if the user session has expired.
+func (p *politeiawww) getSession(w http.ResponseWriter, r *http.Request) (*user.Session, error) {
```

Comment:
Why still store cookies in `p.store`?  Why not use the database for everything?

Reply:
Less work. This approach still harnesses Gorilla's cookie store but the cookies now refer to the actual session in the database.

Reply:
If we wanted to move everything to the db, how much more work would that entail?  Using both the db and Gorilla's cookie store is a confusing setup and could also lead to data invalidation issues.  It should really only be one or the other.

Reply:
I estimate it would take me another 3-4 days. Alternatively, we could also use something like https://github.com/wader/gormstore -- takes 1-2 days to understand and adapt the `politeiawww/userwww.go` code.

Why do you find the combination of a cookie store with cookies that point to database sessions confusing?

Also, what data invalidation issues are you concerned about?

Reply:
I am bit surprised that this turned out to be a contentious issue since I outlined that this is the solution I would implement from the very beginning. See e.g. item 2 and 3 in the [solution proposal](https://github.com/decred/politeia/issues/647#issuecomment-529119436).

Reply:
> Why do you find the combination of a cookie store with cookies that point to database sessions confusing?

Because you're storing the same data in two different locations.  There should only be a single source of truth for the session data unless there's a good technical reason for why we need both.  Storing it two locations adds complexity and mental overhead.  This would also prevent us from being able to run multiple politeiawww instances (#665).

> Alternatively, we could also use something like https://github.com/wader/gormstore 

Don't want to do that.

> I am bit surprised that this turned out to be a contentious issue since I outlined that this is the solution I would implement from the very beginning. 

Reading a spec is not the same as reading code.  Most of the time, I'm not going to understand the full scope of the problem until I'm looking at working code.

Let's get rid of the in-memory cookie store and move everything over to the database.

Reply:
> @al-maisan I'm guessing that you have the email server setup in your politeiawww config file. If you comment out the email server config params then politeiawww will disable email and send the verification token back in the newuser response. Give that a try. You should see the this log message on politeiawww startup: `2019-09-16 08:10:25.203 [INF] PWWW: Email : DISABLED`

Now, the `politeiawwwcli newuser --verify user@example.com user password` command also works

Reply:
> > Why do you find the combination of a cookie store with cookies that point to database sessions confusing?
> 
> Because you're storing the same data in two different locations. There should only be a single source of truth for the session data unless there's a good technical reason for why we need both. Storing it two locations adds complexity and mental overhead. This would also prevent us from being able to run multiple politeiawww instances (#665).
> 
> > Alternatively, we could also use something like https://github.com/wader/gormstore
> 
> Don't want to do that.
> 
> > I am bit surprised that this turned out to be a contentious issue since I outlined that this is the solution I would implement from the very beginning.
> 
> Reading a spec is not the same as reading code. Most of the time, I'm not going to understand the full scope of the problem until I'm looking at working code.
> 
> Let's get rid of the in-memory cookie store and move everything over to the database.

Alright.. It will be a couple of days though .. I am doing this in my spare time.

Reply:
Understood.  No rush.